compiler/repl-service/server/src/main/scala/com/digitalasset/daml/lf/speedy/ReplServiceMain.scala

```diff
@@ -231,12 +243,28 @@ class ReplService(
 
   import ReplService._
 
+  private val compilerConfig = Compiler.Config
+    .Default(majorLanguageVersion)
+    .copy(stacktracing = Compiler.FullStackTrace)
+
+  private[this] def validatePackagesLanguageVersion(pkgMap: Map[PackageId, Package]): Unit = {
+    val invalidPackages =
+      pkgMap.view.mapValues(_.languageVersion.major).filter(_._2 != majorLanguageVersion)
+    if (invalidPackages.nonEmpty) {
+      throw new IllegalArgumentException(
+        "the following packages don't have expected major LF version "
+          + s"${majorLanguageVersion}: ${invalidPackages}"
```

Comment:
we shouldn't print the contents of the packages! the pid on its own would be enough but ideally we would show the package name + version from `package.metadata` if present

Reply:
I actually only print the major versions: `mapValues(_.languageVersion.major)`.

Reply:
🤦 ah my bad I misread